enterprise/api/pagination.py

```diff
@@ -5,8 +5,41 @@
 from collections import OrderedDict
 from urllib.parse import urlparse
 
+from edx_rest_framework_extensions.paginators import DefaultPagination
 from rest_framework.response import Response
 
+from enterprise.toggles import top_down_assignment_real_time_lcm
+
+
+class PaginationWithFeatureFlags(DefaultPagination):
+    """
+    Adds a ``features`` dictionary to the default paginated response
+    provided by edx_rest_framework_extensions. The ``features`` dict
+    represents a collection of Waffle-based feature flags/samples/switches
+    that may be used to control whether certain aspects of the system are
+    enabled or disabled (e.g., feature flag turned on for all staff users but
+    not turned on for real customers/learners).
+    """
+
+    def get_paginated_response(self, data):
+        """
+        Modifies the default paginated response to include ``features`` dict.
+
+        Arguments:
+            self: PaginationWithFeatureFlags instance.
+            data (dict): Results for current page.
+
+        Returns:
+            (Response): DRF response object containing ``features`` dict.
+        """
+        paginated_response = super().get_paginated_response(data)
+        paginated_response.data.update({
+            'features': {
+                'top_down_assignment_real_time_lcm': top_down_assignment_real_time_lcm(),
```

Comment:
Recommendation: add a function to toggles.py to serialize this dictionary so that adding a new feature flag is limited to just updating one file.

Reply:
Great suggestion! I will update this PR after afternoon meetings.